### PR TITLE
Mask sensitive parameters

### PR DESCRIPTION
**Pull request by andykellr**
### Proposed Change

A new option exists for parameters in SourceType, ProcessorType, and DestinationType specs that designates a parameter as sensitive. Sensitive parameter values are not returned to the user. Instead the text `(sensitive)` is used. If the value is left alone, the existing value is preserved. If the value is changed, the new value is used.

Masked in the UI:

Screenshot 2023-06-13 at 1 13 17 PM

Showing the value of the field just shows `(sensitive)`

Screenshot 2023-06-13 at 1 13 38 PM

Masked in REST calls and the command line:

```sh
# curl -u <user>:<pw> <host>/v1/destinations/<name> | jq
...
        {
          "name": "credentials",
          "value": "(sensitive)",
          "sensitive": true
        },
...
```

```sh
# bindplane get destination <name> -o yaml
...
        - name: credentials
          value: (sensitive)
...
```

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: model/parameterized_resource.go [hunk 1]
```diff
@@ -81,3 +113,92 @@ func (s *ParameterizedSpec) updateDependencies(ctx context.Context, kind Kind, s
...
+func (s *ParameterizedSpec) maskSensitiveParameters(ctx context.Context) {
+	maskSensitiveParameters(ctx, s)
+	for i, p := range s.Processors {
+		p.maskSensitiveParameters(ctx)
+		s.Processors[i] = p
+	}
+}
+
+// maskSensitiveParameters will mask sensitive parameters in the spec based on the Sensitive option in the
+// ParameterDefinition. If the specified ResourceType is nil, an error is returned.
```
Comment (shazlehu) [inline model/parameterized_resource.go:126]: Do we need error return value or a nil check here?

**Review by dsvanlani — CHANGES_REQUESTED** [review]
Not sure why but I'm getting `(sensitive)` in my collector config.yaml

```yaml
receivers:
    mysql/source0:
        collection_interval: 60s
        database: null
        endpoint: localhost:3306
        metrics: null
        password: (sensitive)
        transport: tcp
        username: test
    plugin/source0__mysql:
        parameters:
            enable_error_log: true
            enable_general_log: true
            enable_mariadb_audit_log: false
            enable_slow_log: true
            error_log_paths:
                - /var/log/mysqld.log
                - /var/log/mysql/mysqld.log
                - /var/log/mysql/error.log
            general_log_paths:
                - /var/log/mysql/general.log
            slow_query_log_paths:
                - /var/log/mysql/slow*.log
            start_at: end
        path: $OIQ_OTEL_COLLECTOR_HOME/plugins/mysql_logs.yaml
    prometheus/_agent_metrics:
        config:
            scrape_configs:
                - job_name: observiq-otel-collector
                  metric_relabel_configs:
                    - action: keep
                      regex: otelcol_processor_throughputmeasurement_.*
                      source_labels:
                        - __name__
                  scrape_interval: 10s
                  static_configs:
                    - labels:
                        agent: 01H2XQN47XKVRGXDV9WA69E6S7
                        configuration: sensitive
                      targets:
                        - 0.0.0.0:8888
processors:
    batch/_agent_metrics: null
    resourcedetection/source0:
        detectors:
            - system
        system:
            hostname_sources:
                - os
    snapshotprocessor: null
    snapshotprocessor/_d0_logging-0: null
    snapshotprocessor/_s0_source0: null
    throughputmeasurement/_d0_logs_logging-0:
        enabled: true
        sampling_ratio: 1
    throughputmeasurement/_d0_metrics_logging-0:
        enabled: true
        sampling_ratio: 1
    throughputmeasurement/_d1_logs_logging-0:
        enabled: true
        sampling_ratio: 1
    throughputmeasurement/_d1_metrics_logging-0:
        enabled: true
        sampling_ratio: 1
    throughputmeasurement/_s0_logs_source0:
        enabled: true
        sampling_ratio: 1
    throughputmeasurement/_s0_metrics_source0:
        enabled: true
        sampling_ratio: 1
    throughputmeasurement/_s1_logs_source0:
        enabled: true
        sampling_ratio: 1
    throughputmeasurement/_s1_metrics_source0:
        enabled: true
        sampling_ratio: 1
exporters:
    logging/logging: null
    otlphttp/_agent_metrics:
        endpoint: http://127.0.0.1:3001/v1/otlphttp
        headers: {}
        retry_on_failure:
            enabled: true
            initial_interval: 5s
            max_elapsed_time: 30s
            max_interval: 5s
        sending_queue:
            enabled: true
            num_consumers: 1
            queue_size: 60
service:
    pipelines:
        logs/source0__logging-0:
            receivers:
                - plugin/source0__mysql
            processors:
                - resourcedetection/source0
                - throughputmeasurement/_s0_logs_source0
                - snapshotprocessor/_s0_source0
                - throughputmeasurement/_s1_logs_source0
                - throughputmeasurement/_d0_logs_logging-0
                - snapshotprocessor/_d0_logging-0
                - throughputmeasurement/_d1_logs_logging-0
                - snapshotprocessor
            exporters:
                - logging/logging
        metrics/_agent_metrics:
            receivers:
                - prometheus/_agent_metrics
            processors:
                - batch/_agent_metrics
            exporters:
                - otlphttp/_agent_metrics
        metrics/source0__logging-0:
            receivers:
                - mysql/source0
            processors:
                - resourcedetection/source0
                - throughputmeasurement/_s0_metrics_source0
                - snapshotprocessor/_s0_source0
                - throughputmeasurement/_s1_metrics_source0
                - throughputmeasurement/_d0_metrics_logging-0
                - snapshotprocessor/_d0_logging-0
                - throughputmeasurement/_d1_metrics_logging-0
                - snapshotprocessor
            exporters:
                - logging/logging
        
```

File: model/parameterized_resource.go (hunk 1 above)
Comment (andykellr) [reply model/parameterized_resource.go:126]: Sorry, that comment was outdated. It no longer returns an error.

**Review by shazlehu — APPROVED** [review]
Looks good. Also upgraded an agent while testing, everything worked well.

**Pull request closed without merging**